File: app/src/main/java/com/app/rupiksha/adapters/BankListAdapter.java
package com.app.rupiksha.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.app.rupiksha.R;
import com.app.rupiksha.databinding.BanklistLayoutBinding;
import com.app.rupiksha.models.PayoutAccountModel;

import java.util.List;

public class BankListAdapter extends RecyclerView.Adapter<BankListAdapter.ViewHolder> {
    private Context context;
    private List<PayoutAccountModel> list;
    BankListAdapter.OnItemClick onItemClick;

    public BankListAdapter(Context context, List<PayoutAccountModel> stepList, OnItemClick onItemClick) {
        this.context = context;
        this.list = stepList;
        this.onItemClick=onItemClick;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        BanklistLayoutBinding view = DataBindingUtil.bind(LayoutInflater.from(parent.getContext()).inflate(R.layout.banklist_layout, parent, false));
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position)
    {
        if(position==0){
            holder.binding.line2.setVisibility(View.VISIBLE);
        }else{
            holder.binding.line2.setVisibility(View.GONE);
        }
        holder.binding.valueaccountnumber.setText(list.get(position).getAccount());
        holder.binding.valuename.setText(list.get(position).getName());
        holder.binding.valuestatus.setText(list.get(position).getStatus());

        holder.binding.valueaction.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onItemClick.onClick(position);
                list.remove(position);
            }
        });
    }


    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private BanklistLayoutBinding binding;

        public ViewHolder(@NonNull BanklistLayoutBinding itemView) {
            super(itemView.getRoot());
            this.binding = itemView;
        }
    }

    public static interface OnItemClick
    {
        public void onClick(int position);
    }


}

File: app/src/main/java/com/app/rupiksha/adapters/BankListAdapter.kt
package com.app.rupiksha.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.app.rupiksha.R
import com.app.rupiksha.databinding.BanklistLayoutBinding
import com.app.rupiksha.models.PayoutAccountModel

class BankListAdapter(
    private val context: Context,
    private val list: MutableList<PayoutAccountModel>,
    private val onItemClick: OnItemClick
) : RecyclerView.Adapter<BankListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding: BanklistLayoutBinding = DataBindingUtil.inflate(
            LayoutInflater.from(parent.context),
            R.layout.banklist_layout,
            parent,
            false
        )
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (position == 0) {
            holder.binding.line2.visibility = View.VISIBLE
        } else {
            holder.binding.line2.visibility = View.GONE
        }
        
        val model = list[position]
        holder.binding.valueaccountnumber.text = model.account
        holder.binding.valuename.text = model.name
        holder.binding.valuestatus.text = model.status

        holder.binding.valueaction.setOnClickListener {
            onItemClick.onClick(position)
            list.removeAt(position)
            notifyItemRemoved(position)
            notifyItemRangeChanged(position, list.size)
        }
    }

    override fun getItemCount(): Int = list.size

    inner class ViewHolder(val binding: BanklistLayoutBinding) : RecyclerView.ViewHolder(binding.root)

    interface OnItemClick {
        fun onClick(position: Int)
    }
}
